6-AdvancedScenarios/2-call-api-pop/README.md

```diff
@@ -0,0 +1,343 @@
+
+ 1. [Overview](#overview)
+ 1. [Scenario](#scenario)
+ 1. [Contents](#contents)
+ 1. [Prerequisites](#prerequisites)
+ 1. [Setup](#setup)
+ 1. [Registration](#registration)
+ 1. [Running the sample](#running-the-sample)
+ 1. [Explore the sample](#explore-the-sample)
+ 1. [About the code](#about-the-code)
+ 1. [More information](#more-information)
+ 1. [Community Help and Support](#community-help-and-support)
+ 1. [Contributing](#contributing)
+
+## Overview
+
+This sample demonstrates an React SPA calling a Node.js & Express web API that is secured using [Azure Active Directory](https://docs.microsoft.com/azure/active-directory/fundamentals/active-directory-whatis) (Azure AD). The SPA project is secured with the [Microsoft Authentication Library for React (Preview)](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-react) (MSAL React), while the web API is secured with [passport-azure-ad](https://github.com/AzureAD/passport-azure-ad).
+
+This sample demonstrates the [Proof of Possession](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/access-token-proof-of-possession.md) (PoP) authentication scheme. This authentication scheme cryptographically binds the access tokens to the browser and client application from which they are requested, meaning they cannot be used from a different application or device. This effectively prevents **token replay** attacks. The resource server that accepts the PoP token (i.e. a web API) needs to be able to decipher the incoming request for the PoP authentication scheme to work properly.
+
+## Scenario
+
+1. The client React SPA uses **MSAL React** to sign-in and obtain a JWT access token from **Azure AD**.
+1. The client React SPA wraps the access token in a secure *envelope* and stamps it with its signature.
+1. The access token then is used with *PoP* authentication scheme to authorize the user to call the Node.js & Express web API protected by **Azure AD**.
+1. The Node.js & Express web API decrypts the secure envelope to obtain the access token.
+1. Once the access token is validated, the Node.js & Express web API responds with the protected resource.
+
+![Overview](./ReadmeFiles/topology.png)
+
+## Contents
+
+| File/folder                         | Description                                                |
+|-------------------------------------|------------------------------------------------------------|
+| `SPA/src/authConfig.js`             | Authentication parameters for SPA project reside here.     |
+| `SPA/src/index.js`                  | MSAL React is initialized here.                            |
+| `API/authConfig.json`               | Authentication parameters for API project reside here.     |
+| `API/app.js`                        | Application entry. passport-azure-ad is initialized here.  |
+| `API/utils/validateToken.js`        | Contains utility methods for validating PoP tokens.        |
```

Comment:
>validateToken [](http://example.com/codeflow?start=13&length=13)

also call out the is the a file where the key wrap is executed, fetch.js?

Reply:
added